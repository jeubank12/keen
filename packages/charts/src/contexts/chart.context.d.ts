import React from 'react';
import { Dimension, Margins, Theme, ScaleSettings } from '../types';
export declare type ChartContextType = {
  theme: Theme;
  svgDimensions: Dimension;
  margins: Margins;
  xScaleSettings: ScaleSettings;
  yScaleSettings: ScaleSettings;
  data: object[];
};
declare const ChartContext: React.Context<{}>;
export default ChartContext;
