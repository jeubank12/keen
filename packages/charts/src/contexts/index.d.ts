import ChartContext, { ChartContextType } from './chart.context';
import ScaleContext, { ScaleContextType } from './scale.context';
export { ChartContext, ScaleContext, ScaleContextType, ChartContextType };
