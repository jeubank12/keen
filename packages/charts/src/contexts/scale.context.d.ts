import React from 'react';
import { ScaleBand, ScaleLinear } from 'd3-scale';
export declare type ScaleContextType = {
  xScale: ScaleBand<string> | ScaleLinear<number, number>;
  yScale: ScaleLinear<number, number> | ScaleBand<string>;
};
declare const ScaleContext: React.Context<{}>;
export default ScaleContext;
