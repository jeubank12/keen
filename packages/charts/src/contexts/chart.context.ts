import React from 'react';
import { ScaleSettings } from '@keen.io/charts-utils';

import { Dimension, Margins, Theme } from '../types';

export type ChartContextType = {
  theme: Theme;
  svgDimensions: Dimension;
  margins: Margins;
  xScaleSettings: ScaleSettings;
  yScaleSettings: ScaleSettings;
  data: Record<string, any>[];
};

const ChartContext = React.createContext({});

export default ChartContext;
