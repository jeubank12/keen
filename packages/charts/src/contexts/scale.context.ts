import React from 'react';
import { ScaleBand, ScaleLinear } from 'd3-scale';

export type ScaleContextType = {
  xScale: ScaleBand<string> | ScaleLinear<number, number>;
  yScale: ScaleLinear<number, number> | ScaleBand<string>;
};

const ScaleContext = React.createContext({});

export default ScaleContext;
