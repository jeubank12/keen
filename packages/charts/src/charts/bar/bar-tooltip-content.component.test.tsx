import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { ChartContext } from '../../contexts';

import BarTooltip from './bar-tooltip-content.component';

const render = (overProps: any = {}) => {
  const data = [
    { name: 'Windows', users: 3, licenses: 52, shops: 12 },
    { name: 'MacOS', users: 19, licenses: 82, shops: 15 },
    { name: 'Linux', users: 20, licenses: -15, shops: 23 },
    { name: 'Android', users: 63, licenses: -15, shops: -30 },
  ];
  const keys = ['users', 'licenses', 'shops'];
  const selectors = [
    {
      color: '#85B4C3',
      selector: [3, 'users'],
    },
  ];
  const formatTooltip = value => `$${value}`;
  const props = {
    data,
    keys,
    selectors,
    groupMode: 'normal',
    stackMode: 'grouped',
    isList: false,
    formatTooltip,
    ...overProps,
  };

  const wrapper = rtlRender(
    <ChartContext.Provider
      value={{ theme: { tooltip: { labels: { typography: {} } } } }}
    >
      <BarTooltip {...props} />
    </ChartContext.Provider>
  );

  return {
    wrapper,
    props,
  };
};

test('formats tooltip value', () => {
  const {
    wrapper: { getByText },
    props: { data, formatTooltip },
  } = render();
  const value = data[3].users;
  const formattedValue = formatTooltip(value);

  expect(getByText(formattedValue)).toBeInTheDocument();
});

test('formats toolip displayed as a list', () => {
  const data = [
    {
      'keen.key': '2019-12-01T00:00:00.000Z',
      'Love, Anger, Madness Edwidge Danticat': 105,
      'Game of Thrones George R. R. Martin': 79,
      'Harry Potter J.K. Rowling': 102,
      'It Stephen King': 47,
      'The Shining Stephen King': 0,
    },
    {
      'keen.key': '2020-01-01T00:00:00.000Z',
      'Love, Anger, Madness Edwidge Danticat': 95,
      'Game of Thrones George R. R. Martin': 719,
      'Harry Potter J.K. Rowling': 112,
      'It Stephen King': 2059,
      'The Shining Stephen King': 6,
    },
    {
      'keen.key': '2020-02-01T00:00:00.000Z',
      'Love, Anger, Madness Edwidge Danticat': 2,
      'Game of Thrones George R. R. Martin': 11,
      'Harry Potter J.K. Rowling': 1,
      'It Stephen King': 45,
      'The Shining Stephen King': 0,
    },
  ];
  const keys = [
    'Love, Anger, Madness Edwidge Danticat',
    'Game of Thrones George R. R. Martin',
    'Harry Potter J.K. Rowling',
    'It Stephen King',
    'The Shining Stephen King',
  ];
  const disabledKeys = [];
  const selectors = [
    {
      selector: [2, 'Love, Anger, Madness Edwidge Danticat'],
      color: '#85B4C3',
    },
    { selector: [2, 'Game of Thrones George R. R. Martin'], color: '#CB5623' },
    { selector: [2, 'Harry Potter J.K. Rowling'], color: '#E29B1E' },
    { selector: [2, 'It Stephen King'], color: '#487650' },
    { selector: [2, 'The Shining Stephen King'], color: '#27566D' },
  ];
  const stackMode = 'percent';
  const groupMode = 'stacked';
  const isList = true;

  const { wrapper } = render({
    data,
    keys,
    disabledKeys,
    selectors,
    stackMode,
    groupMode,
    isList,
  });
  expect(wrapper).toMatchSnapshot();
});
