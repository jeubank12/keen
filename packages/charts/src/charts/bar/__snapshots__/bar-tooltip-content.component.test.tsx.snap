// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formats toolip displayed as a list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="bar-tooltip"
      >
        <ul
          class="sc-fznyAO jnstRf"
        >
          <li
            class="sc-fznKkj cEsaz"
          >
            <div
              class="sc-fznZeY LHfpP"
              color="#85B4C3"
            />
            <div
              class="sc-AxjAm ldjeaj"
            >
              $2 (3.39%)
            </div>
          </li>
          <li
            class="sc-fznKkj cEsaz"
          >
            <div
              class="sc-fznZeY eIskkf"
              color="#CB5623"
            />
            <div
              class="sc-AxjAm ldjeaj"
            >
              $11 (18.64%)
            </div>
          </li>
          <li
            class="sc-fznKkj cEsaz"
          >
            <div
              class="sc-fznZeY jFjnNA"
              color="#E29B1E"
            />
            <div
              class="sc-AxjAm ldjeaj"
            >
              $1 (1.69%)
            </div>
          </li>
          <li
            class="sc-fznKkj cEsaz"
          >
            <div
              class="sc-fznZeY uRBmM"
              color="#487650"
            />
            <div
              class="sc-AxjAm ldjeaj"
            >
              $45 (76.27%)
            </div>
          </li>
          <li
            class="sc-fznKkj cEsaz"
          >
            <div
              class="sc-fznZeY kPUROk"
              color="#27566D"
            />
            <div
              class="sc-AxjAm ldjeaj"
            >
              $0 (0.00%)
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="bar-tooltip"
    >
      <ul
        class="sc-fznyAO jnstRf"
      >
        <li
          class="sc-fznKkj cEsaz"
        >
          <div
            class="sc-fznZeY LHfpP"
            color="#85B4C3"
          />
          <div
            class="sc-AxjAm ldjeaj"
          >
            $2 (3.39%)
          </div>
        </li>
        <li
          class="sc-fznKkj cEsaz"
        >
          <div
            class="sc-fznZeY eIskkf"
            color="#CB5623"
          />
          <div
            class="sc-AxjAm ldjeaj"
          >
            $11 (18.64%)
          </div>
        </li>
        <li
          class="sc-fznKkj cEsaz"
        >
          <div
            class="sc-fznZeY jFjnNA"
            color="#E29B1E"
          />
          <div
            class="sc-AxjAm ldjeaj"
          >
            $1 (1.69%)
          </div>
        </li>
        <li
          class="sc-fznKkj cEsaz"
        >
          <div
            class="sc-fznZeY uRBmM"
            color="#487650"
          />
          <div
            class="sc-AxjAm ldjeaj"
          >
            $45 (76.27%)
          </div>
        </li>
        <li
          class="sc-fznKkj cEsaz"
        >
          <div
            class="sc-fznZeY kPUROk"
            color="#27566D"
          />
          <div
            class="sc-AxjAm ldjeaj"
          >
            $0 (0.00%)
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
